lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -338,35 +346,43 @@ def _run_batch(self, parent_job: Job, params):
 
         return child_jobs
 
-    def run_batch(
-        self, params, batch_params, as_admin=False
-    ) -> Dict[str, Union[Job, List[str]]]:
-        """
-        :param params: List of RunJobParams (See Spec File)
-        :param batch_params: List of Batch Params, such as wsid (See Spec file)
-        :param as_admin: Allows you to run jobs in other people's workspaces
-        :return: A list of condor job ids or a failure notification
-        """
+    @staticmethod
+    def _check_batch_params(params):
         if type(params) != list:
             raise IncorrectParamsException("params must be a list")
-        wsid = batch_params.get(_WORKSPACE_ID)
-        meta = batch_params.get(_META)
 
+        for item in params:
+            if not item:
+                raise MissingRunJobParamsForBatchException()
+
+    def _preflight_batch(self, params: List[Dict], as_admin=False, parent_wsid=None):
+        self._check_batch_params(params)
         if as_admin:
             self.sdkmr.check_as_admin(requested_perm=JobPermissions.WRITE)
         else:
-            # Make sure you aren't running a job in someone elses workspace
-            self._check_workspace_permissions(wsid)
-            # this is very odd. Why check the parent wsid again if there's no wsid in the job?
-            # also, what if the parent wsid is None?
-            # also also, why not just put all the wsids in one list and make one ws call?
-            wsids = [job_input.get(_WORKSPACE_ID, wsid) for job_input in params]
+            wsids = [
+                job_param.get("wsid") for job_param in params if job_param.get("wsid")
+            ]
+            if parent_wsid:
+                wsids.append(parent_wsid)
             self._check_workspace_permissions_list(wsids)
-
-        self._add_job_requirements(params, bool(as_admin))  # as_admin checked above
+        self._handle_job_requirements(params=params, as_admin=as_admin)
         self._check_job_arguments(params, has_parent_job=True)
 
-        parent_job = self._create_parent_job(wsid=wsid, meta=meta)
+    def run_batch(
+        self, params: List[Dict], batch_params: Dict, as_admin: bool = False
+    ) -> Dict[str, List[str]]:
+        """
+        :param params: List of RunJobParams (See Spec File)
+        :param batch_params: List of Batch Params, such as wsid (See Spec file)
+        :param as_admin: Allows you to run jobs in other people's workspaces
+        :return: A list of condor job ids or a failure notification
```

Comment:
These are ee2 job IDs not condor job IDs, right? And it's a dict not a list

Reply:
Fixed